### PR TITLE
Miscellaneous cleanup

### PR DESCRIPTION
**Pull request by archigup**
Add doc/test for memory statistics and some cleanup.
Adds a flag to uncrustify which is needed for coreJSON.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -95,3 +95,26 @@ jobs:
           path: ./build/coverage.info
           branch-min-coverage: 100
           line-min-coverage: 99
+  test-memory-statistics:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/checkout@v2
```
Comment (aggarw13) [inline .github/workflows/test.yml:102]:
```suggestion
      - uses: actions/checkout@v2
```

File: README.md [hunk 2]
```diff
@@ -1,11 +1,20 @@
...
+  standard defined in [uncrustify.cfg](formatting/uncrustify.cfg).
+* **Doxygen** - Validates that the doxygen manual of the FreeRTOS library can be built without
+  warnings.
+* **Spellings** - Checks spellings across all files of the FreeRTOS library repository. Each
+  FreeRTOS library repository should have a **lexicon.txt** file.
+* **Coverage Cop** - Enforces that the unit tests of a FreeRTOS library meet the minimum thresholds
+  branch and line coverages. The **lcov** coverage output from running unit tests should be
+  available before using this action.
+* **Memory Statistics** - Generates the library size tables used in FreeRTOS library
+  documentation.
```
Comment (aggarw13) [inline README.md:20]:
```suggestion
* **Memory Statistics** - Generates table of memory estimates of library files that is used in FreeRTOS library documentation. The memory estimates are generated by building the library with ARM GCC toolchain.

```

File: doxygen/action.yml [hunk 3]
```diff
@@ -5,18 +5,21 @@ inputs:
     description: 'Path to repository folder to run doxygen generation in.'
     required: false
     default: ./
+  doxygen_link:
+    description: 'Link to doxygen tar.gz to use 9 (default version 1.8.20).'
```
Comment (aggarw13) [inline doxygen/action.yml]:
```suggestion
    description: 'Download link to doxygen tar.gz (default version 1.8.20).'
```

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -95,3 +95,26 @@ jobs:
           path: ./build/coverage.info
           branch-min-coverage: 100
           line-min-coverage: 99
+  test-memory-statistics:
+    runs-on: ubuntu-latest
```
Comment (aggarw13) [inline .github/workflows/test.yml:99]: Can you either print or upload the output file of this action as an artifact for verifying that this action does provide a table in the output file?

**Review by aggarw13 — APPROVED** [review]

**Review by dan4thewin — APPROVED** [review]

**Review by yourslab — APPROVED** [review]

**Pull request merged**